### PR TITLE
Support sqlite

### PR DESCRIPTION
**Pull request by boosh**
**Description of the change**
Supports using sqlite as a DB backend.

**Benefits**
Supports sqlite

**Possible drawbacks**
Only added for Debian 8. Debian 6 still needs updating.

**Applicable issues**
#143 

**Additional information**
Tested with an sqlite configuration. Needs more testing with the default docker-compose file (i.e. with mariadb)

### CODE REVIEW

File: 8/debian-10/rootfs/app-entrypoint.sh [hunk 1]
```diff
@@ -45,6 +45,18 @@ replace_in_file() {
...
+      local -r db_database="${DB_DATABASE:-/app/database.sqlite}"
+
+      if [[ ! -f "$db_database" ]]; then
+        log "Creating sqlite database at $db_database"
+        sqlite3 "$db_database" ""
+      fi
+
+      return
+    fi
+
```
Comment (rafariossaa) [inline 8/debian-10/rootfs/app-entrypoint.sh:59]: As this is a `wait_...` function, I would not add the creation here.
I would suggest to move the logic to `setup_db`, and only call `wait_for_db` if the DB_CONNECTION is not sqlite

What do you think?

Comment (boosh) [reply 8/debian-10/rootfs/app-entrypoint.sh:59]: Hey yeah I'm easy. Feel free to edit it how you like.

Comment (boosh) [reply 8/debian-10/rootfs/app-entrypoint.sh:59]: Actually that's not possible without a more complicated PR. The current code always calls `wait_for_db`. There's some conditional before calling `setup_db`

**Comment by rafariossaa** [comment]
Hi,
  ok, I am creating an internal task to address this.
  We will come back as soon as we have news.

**Comment by boosh** [comment]
Hi, any chance you can merge this and refactor it later please?

File: 8/debian-10/rootfs/app-entrypoint.sh [hunk 2]
```diff
@@ -45,6 +45,18 @@ replace_in_file() {
...
+    if [[ "$db_connection" == "sqlite" ]]; then
+      local -r db_database="${DB_DATABASE:-/app/database.sqlite}"
+
+      if [[ ! -f "$db_database" ]]; then
+        log "Creating sqlite database at $db_database"
+        sqlite3 "$db_database" ""
+      fi
+
+      return
+    fi
```
Comment (marcosbc) [inline 8/debian-10/rootfs/app-entrypoint.sh:58]: Could you check the indentation for this block, and ensure that it is consistent with the 4-whitespace rule we are following throughout the file?

File: 8/debian-10/rootfs/app-entrypoint.sh [hunk 3]
```diff
@@ -45,6 +45,18 @@ replace_in_file() {
 # Returns: none
 #########################
 wait_for_db() {
+    local -r db_connection="${DB_CONNECTION:-mysql}"
```
Comment (marcosbc) [inline 8/debian-10/rootfs/app-entrypoint.sh:48]: I'd recommend to rename this environment variable to `DB_TYPE` to be consistent with what we are using in other container images:

```suggestion
    local -r db_connection="${DB_TYPE:-mysql}"
```

**Review by marcosbc — COMMENTED** [review]
I agree with you, let's try and get this PR merged! Could you take a look at my comments?

File: 8/debian-10/rootfs/app-entrypoint.sh (hunk 3 above)
Comment (boosh) [reply 8/debian-10/rootfs/app-entrypoint.sh:48]: That env var appears to be from [laravel](https://laravel.com/docs/8.x/database#sqlite-configuration)

Comment (marcosbc) [reply 8/debian-10/rootfs/app-entrypoint.sh:48]: Good point!

**Review by marcosbc — APPROVED** [review]
LGTM! Thanks for contributing this!

**Pull request merged**